main.go

```diff
@@ -139,8 +144,16 @@ var cmdInstall = &cli.Command{
 		if vers == "latest" || vers == "beta" {
 			stable := vers == "latest"
 			latest, err := migrations.LatestDistVersion(c.Context, fetcher, "go-ipfs", stable)
-			if err != nil {
-				stump.Fatal("error resolving %q: %s", vers, err)
+			for i := c.Int("retries"); i > 0; i-- {
+				if err == nil {
+					break
+				}
+				if i > 1 { // On the last attempt we want a failed attempt to be logged as fatal
+					stump.Error("error resolving %q: %s", vers, err)
+				} else {
+					stump.Fatal("error resolving %q: %s", vers, err)
+				}
+				latest, err = migrations.LatestDistVersion(c.Context, fetcher, "go-ipfs", stable)
```

Comment:
There really isn't any point in re-trying when downloading from IPFS, only over HTTP. This flag should probably be piped into `NewHttpFetcher`.